File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        freq = {}
        for c in s:
            if c in freq:
                freq[c] += 1
            else:
                freq[c] = 1
        
        odd = 0
        for cnt in freq.values():
            if cnt % 2 == 1:
                odd += 1
        
        res = len(s)
        if odd > 0:
            res -= (odd - 1)

        return res

File: 0409-longest-palindrome/0409-longest-palindrome.go
func longestPalindrome(s string) int {
    freq := map[rune]int{}
    for _, c := range s {
        freq[c]++
    }

    ans := 0
    odd := 0
    for _, total := range freq {
        if total % 2 == 0 {
            ans += total
        } else {
            ans += (total - 1)
            odd++
        }
    }

    if odd > 0 {
        ans++
    }

    return ans
}